### ISSUE TITLE
[Android] Download are cancelled with no reason

### ISSUE THREAD

**Issue by DelattreAdixon** [SimonSimCity/Xamarin-CrossDownloadManager#77]
Fork reproducing the issue : 
https://github.com/DelattreAdixon/Xamarin-CrossDownloadManager
Launch the DownloadExample.Droid project, and look at the ouput. Some files are marked as 'Cancelled' :

```
...
[Property changed] Status -> CANCELED
content://downloads/all_downloads/3535
[Property changed] Status -> CANCELED
content://downloads/all_downloads/3535
[DownloadManager] Remove -> New items: 0 at -1 || Old items: 1 at -1
[Property changed] Status -> CANCELED
content://downloads/all_downloads/3536
[Property changed] Status -> CANCELED
content://downloads/all_downloads/3536
...
```

Looking at the Xamarin

### Steps to reproduce

1. Add several files in the download queue (more than 30 files), and start them

2. Some files are not downloaded. When checking the status for each download, we see that some files have been cancelled (though not cancelled intentionally), with no explicit reason.

### Expected behavior
All files in the download queue should be downloaded

### Actual behavior
Some download are canceled 

### Configuration

**Platform:** 
- [ ] iOS
- [X] Android
- [ ] UWP
- [ ] Other

**Device:** 
- [X] Simulator
- [X] Real device (please specify)

**Comment by SimonSimCity** [comment]
Thanks very much for the fork, which made it very easy for me to track down the problem. I've created a fix on my `develop` branch. Please review this change, merge it into your fork and verify that this problem is fixed.

If you don't respond within one week I consider this issue as fixed 😉

**Comment by SimonSimCity** [comment]
I've now published the version `1.3.6` which should fix this bug. Please tell me if it should still occur.
